File: backend/internal/domain/post.go
package domain

import "time"

type Post struct {
	ID          string     `json:"id"`
	PublisherID string     `json:"publisher_id"`
	Title       string     `json:"title"`
	Subtitle    *string    `json:"subtitle"`
	Slug        string     `json:"slug"`
	Content     *string    `json:"post_content"`
	Status      PostStatus `json:"status"`
	NumRays     int64      `json:"num_rays"`
	NumComments int64      `json:"num_comments"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
	DeletedAt   *time.Time `json:"deleted_at"`

	Publisher  *Profile
	CoverImage *PostImage
	Video      *PostVideo
}

type PostStatus string

const (
	Draft     PostStatus = "draft"
	Published PostStatus = "published"
	Archived  PostStatus = "archived"
)

File: the-daily-sunshine-frontend/src/types/Post.ts
import type { PostImage } from "./PostImage";
import type { Profile } from "./Profile";

const PostStatuses = {
    Draft: "draft",
    Published: "published",
    Archived: "archived"
} as const;

export type PostStatus = typeof PostStatuses[keyof typeof PostStatuses];

export interface Post {
    id: string;
    publisher_id: string;
    title: string;
    subtitle?: string;
    slug: string;
    content?: string;
    status: PostStatus;
    num_rays: number;
    num_comments: number;
    created_at: Date;
    updated_at: Date;
    deleted_at?: Date;

    Publisher: Profile;
    CoverImage?: PostImage;
    Images: PostImage[];
}

export interface NewPostRequest {
    title: string;
    subtitle?: string;
    content?: string;
    status: PostStatus;
}



